feat: Make sure that storm caching settings are resonable (#285)
awslabs/aws-encryption-sdk-specification: framework/storm-tracking-cryptographic-materials-cache.md

- [Grace Period](#grace-period)
- [Grace Interval](#grace-interval)
- [FanOut](#fanout)
- [Inflight TTL](#inflight-ttl)

> - [sleepMilli](#sleepmilli).

Author: ```suggestion
- [sleepMilli](#sleepmilli)
```

- [Grace Period](#grace-period)
- [Grace Interval](#grace-interval)
- [FanOut](#fanout)
- [Inflight TTL](#inflight-ttl)
- [sleepMilli](#sleepmilli).

The implementation MUST instantiate a [Local CMC](local-cryptographic-materials-cache.md)
to do the actual caching.

The settings need to be consistent.

> Here are examples of ambagious or in-consistent settings:

Reviewer: ```suggestion
Here are examples of ambiguous or inconsistent settings:
```

> All grace intervals that exceeds a given grace period are the same because only 1 attempt is made per grace interval and the grace period will end before the next interval.

Reviewer: ```suggestion
A grace interval that exceeds the grace period is inconsistent because only one attempt is made per grace interval and the grace period will end before the next interval.
```

> All in flight TTLs that exceeds a given grace period are the same because the grace period will expire before the in flight TTL.

Reviewer: ```suggestion
An in flight TTL that exceeds the grace period is inconsistent because the grace period will expire before the in flight TTL.
```

> All in flight TTLs that exceeds a given grace period are the same because the grace period will expire before the in flight TTL.

Reviewer: ```suggestion
An in flight TTL that exceeds the grace period is inconsistent because the grace period will expire before the in flight TTL.
```

> All in flight TTLs that are less than a given grace interval are the same because only 1 attempt is made per grace interval and even if the in flight TTL expires before the interval another attempt should not start.

Reviewer: ```suggestion
An in flight TTL that is less than the grace interval is inconsistent because only one attempt is made per grace interval and even if the in flight TTL expires before the interval another attempt should not start.
```
